src/clients/SpokePoolClient.ts

```diff
@@ -305,7 +324,30 @@ export class SpokePoolClient extends BaseAbstractClient {
 
       return updatedDeposit;
     }
-    assert(false); // v3 is coming.
+
+    const v3SpeedUps = this.speedUps[depositor]?.[depositId]?.filter(isV3SpeedUp);
+    const maxSpeedUp = v3SpeedUps?.reduce((prev, current) => {
+      assert(isV3SpeedUp(prev) && isV3SpeedUp(current)); // tsc hinting.
+      return prev.updatedOutputAmount.lt(current.updatedOutputAmount) ? prev : current;
+    });
+    assert(!isDefined(maxSpeedUp) || isV3SpeedUp(maxSpeedUp)); // tsc hinting.
```

Comment:
Similar to the above comment, I think we should be able to get the right types if we help typescript know the input and output types of the function passed to reduce, right? It should infer

Reply:
As above: https://github.com/across-protocol/sdk-v2/pull/501#discussion_r1481529634

Reply:
Resolved: https://github.com/across-protocol/sdk-v2/pull/501/commits/3e260c134a0f9e49e0cd986bf3e708e232260c88